marshal.go

```diff
@@ -109,7 +109,13 @@ func marshal(p *Params, xv reflect.Value, pt *requestType) error {
 		return errgo.Mask(err)
 	}
 	p.Request.URL.Path = path
-	p.Request.URL.RawQuery = p.Request.Form.Encode()
+	rawQuery := p.Request.Form.Encode()
```

Comment:
if q := p.Request.Form.Encode(); q != "" && p.Request.URL.RawQuery != "" {
             p.Request.URL.RawQuery += "&" + q
    } else {
             p.Request.URL.RawQuery = q
    }

? (one less allocation in one branch and arguably marginally simpler)